### PR TITLE
Migrate to Vite

### LINKED ISSUE #141: Migrate to Vite
😉

### PR DESCRIPTION
**Pull request by swouf**
Migrate to Vite for building the app. Upgrades some dependencies on the way.

Close #141

### CODE REVIEW

File: .github/workflows/deploy-prod.yml [hunk 1]
```diff
@@ -1,14 +1,13 @@
-name: Deploy to production environment
+name: Deploy to staging environment
```
Comment (spaenleh) [inline .github/workflows/deploy-prod.yml:1]: You have a typo here, it should be `production`
```suggestion
name: Deploy to production environment
```

File: .github/workflows/deploy-stage.yml [hunk 2]
```diff
@@ -6,10 +6,9 @@ on:
   repository_dispatch:
     types: [staging-deployment]
 
-# This workflow is made up of one job that calls the reusable workflow in graasp-deploy
 jobs:
-  deploy-app-to-staging:
-    name: Deploy to staging
+  deploy-app:
+    name: Deploy to dev
```
Comment (spaenleh) [inline .github/workflows/deploy-stage.yml:11]:
```suggestion
    name: Deploy to staging
```

File: package.json [hunk 3]
```diff
@@ -26,117 +26,114 @@
...
-    "start:test": "env-cmd -f ./.env.test react-scripts --max_old_space_size=4096 -r @cypress/instrument-cra start",
-    "start:ci": "react-scripts --max_old_space_size=4096 -r @cypress/instrument-cra start",
-    "build": "react-scripts --max_old_space_size=4096 build",
-    "build:dev": "env-cmd -f ./.env.dev react-scripts --max_old_space_size=4096 build",
-    "test:cra": "react-scripts test",
-    "eject": "react-scripts eject",
+    "dev:mock": "VITE_MOCK_API=true && yarn vite",
+    "dev": "yarn vite",
+    "start:test": "yarn vite --mode test",
+    "build": "NODE_OPTIONS=--max_old_space_size=16384 && yarn vite build",
```
Comment (spaenleh) [inline package.json:72]: You should not need that option. Try removing it, and checking your versions of sdk, immutable etc to see if they match.

File: package.json [hunk 4]
```diff
@@ -26,117 +26,114 @@
...
+    "preview": "yarn vite preview",
```
Comment (spaenleh) [inline package.json:73]: I think you can omit the `yarn` for these commands, `vite build` should be enough for example.

File: package.json [hunk 5]
```diff
@@ -26,117 +26,114 @@
...
-    "@types/react-dom": "18.0.11",
-    "@types/uuid": "^9.0.1",
-    "@typescript-eslint/eslint-plugin": "5.56.0",
-    "@typescript-eslint/parser": "5.56.0",
+    "@types/react-dom": "18.2.7",
+    "@types/uuid": "^9.0.2",
+    "@typescript-eslint/eslint-plugin": "5.62.0",
+    "@typescript-eslint/parser": "5.62.0",
+    "@vitejs/plugin-react": "4.0.3",
     "codacy-coverage": "3.4.0",
```
Comment (spaenleh) [inline package.json:109]: I don't think we use this anymore.

File: package.json [hunk 6]
```diff
@@ -26,117 +26,114 @@
...
-    "eslint-plugin-react": "~7.32.2",
+    "eslint-plugin-react": "~7.33.0",
     "eslint-plugin-react-hooks": "~4.6.0",
-    "glob": "~9.3.2",
-    "html-webpack-plugin": "~5.5.0",
+    "html-webpack-plugin": "~5.5.3",
     "husky": "~8.0.3",
-    "interpolate-html-plugin": "4.0.0",
-    "json-server": "0.17.3",
     "npm-run-all": "4.1.5",
```
Comment (spaenleh) [inline package.json:128]: I think this could be removed also.

**Review by spaenleh — CHANGES_REQUESTED** [review]
Very nice ! ⭐ Just a few comments, and some typos here and there.

File: .github/workflows/deploy-dev.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      - main
+
+  # Allows to run the workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  deploy-app:
+    name: Deploy to dev
+    runs-on: ubuntu-latest
+    environment: development
```
[the remark this answers is not in the document]
Comment (swouf) [reply .github/workflows/deploy-dev.yml:17]: Okay, I didn't know about that actually. I'll keep it then and will add the secrets to the development env.

**Comment by spaenleh** [comment]
For your tests, you might want to lower the typing speed, it looks like some keystrokes are not sent to the text editor.

**Review by spaenleh — APPROVED** [review]
Looks good, let me know if the tests pass ✅

**Pull request merged**